dev/buildtool/git_support.py

```diff
@@ -1150,37 +1080,33 @@ def determine_git_repository_spec(self, git_dir):
                              origin=origin_url,
                              upstream=remote_urls.get('upstream'))
 
-  def collect_repository_summary(self, git_dir, base_commit_id=None):
-    """Collects RepsitorySummary from local repository directory."""
+  def collect_repository_summary(self, git_dir, start_commit_id=None):
```

Comment:
This is the main change in this commit.

Most callers only require the returned latest `tag_commit_id` and its `tag` to work with.
They don't need a list of commit messages.
They invoke `collect_repository_summary` without a `start_commit_id`.

`build_changelog` is the only different caller in our current release process/use of buildtool.
`build_changelog` will:
1. invoke `collect_repository_summary` with a `start_commit_id` from the previous BOM which is likely different to the latest tag's SHA.
2. use the returned tag on a service in the changelog
3. use the returned list of commit messages in the changelog